### ISSUE TITLE
lightrec: out of bounds read

### ISSUE THREAD

**Issue by notaz** [libretro/pcsx_rearmed#696]
this can crash under some conditions
```
==2866189== Thread 4:
==2866189== Invalid read of size 4
==2866189==    at 0x1DF113: lightrec_emit_eob (emitter.c:95)
==2866189==    by 0x1DF278: lightrec_rec_opcode (emitter.c:2576)
==2866189==    by 0x1E66E5: lightrec_compile_block (lightrec.c:1409)
==2866189==    by 0x1F057D: lightrec_compile_list (recompiler.c:141)
==2866189==    by 0x1F057D: lightrec_recompiler_thd (recompiler.c:192)
==2866189==    by 0x4A9D608: start_thread (pthread_create.c:477)
==2866189==    by 0x4C14132: clone (clone.S:95)
==2866189==  Address 0x120fffd4 is 0 bytes after a block of size 20 alloc'd
==2866189==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2866189==    by 0x1E87E5: lightrec_malloc (memmanager.c:67)
==2866189==    by 0x1E6E39: lightrec_disassemble (lightrec.c:1190)
==2866189==    by 0x1E6E39: lightrec_precompile_block (lightrec.c:1230)
==2866189==    by 0x1E6E39: lightrec_get_block (lightrec.c:637)
==2866189==    by 0x1E7200: get_next_block_func (lightrec.c:662)
==2866189==    by 0x1E7364: lightrec_execute (lightrec.c:1602)
==2866189==    by 0x1A2CED: lightrec_plugin_execute_internal (plugin.c:468)
==2866189==    by 0x194623: psxExecuteBios (r3000a.c:240)
==2866189==    by 0x194623: psxReset (r3000a.c:71)
==2866189==    by 0x2966CC: SysReset (main.c:767)
==2866189==    by 0x16E310: main (main.c:646)

==2866189== Thread 5:
==2866189== Invalid read of size 4
==2866189==    at 0x1DFFD1: rec_b (emitter.c:253)
==2866189==    by 0x1E057A: rec_BEQ (emitter.c:329)
==2866189==    by 0x1DF278: lightrec_rec_opcode (emitter.c:2576)
==2866189==    by 0x1E66E5: lightrec_compile_block (lightrec.c:1409)
==2866189==    by 0x1F057D: lightrec_compile_list (recompiler.c:141)
==2866189==    by 0x1F057D: lightrec_recompiler_thd (recompiler.c:192)
==2866189==    by 0x4A9D608: start_thread (pthread_create.c:477)
==2866189==    by 0x4C14132: clone (clone.S:95)
==2866189==  Address 0x122c3db4 is 0 bytes after a block of size 84 alloc'd
==2866189==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==2866189==    by 0x1E87E5: lightrec_malloc (memmanager.c:67)
==2866189==    by 0x1E6E39: lightrec_disassemble (lightrec.c:1190)
==2866189==    by 0x1E6E39: lightrec_precompile_block (lightrec.c:1230)
==2866189==    by 0x1E6E39: lightrec_get_block (lightrec.c:637)
==2866189==    by 0x1E7200: get_next_block_func (lightrec.c:662)
==2866189==    by 0x48520DC: ???
```

**Comment by pcercuei** [comment]
I think there is (at least) one race left in the threaded code and this is most likely what you're seeing here. I've been chasing it for a while, sadly I didn't find it yet :(

In the meantime I believe it would make sense to disable the threaded compiler.

**Comment by pcercuei** [comment]
Fixed in https://github.com/pcercuei/lightrec/commit/7a439c3 and https://github.com/pcercuei/lightrec/commit/d754e25.

**Comment by notaz** [comment]
This is now fixed along with pcercuei/lightrec#25.
